app/utils/encryptable_transfer.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class EncryptableTransfer
+  def transfer(encryptable, receiver, sender)
+    transfer_password = new_transfer_password
+    encryptable.encrypt(transfer_password)
+
+    encryptable.update!(
+      folder: inbox_folder(receiver),
+      sender_id: sender.id,
+      receiver_id: receiver.id,
+      encrypted_transfer_password: encrypted_transfer_password(transfer_password, receiver)
+    )
+    encryptable
+  end
+
+  def receive(encryptable, private_key, personal_team_password)
+    encryptable.decrypt_transfered(private_key)
+
+    encryptable.update!(encrypted_transfer_password: nil,
+                        receiver_id: nil,
+                        sender_id: nil)
+
+    encryptable.encrypt(personal_team_password)
```

Comment:
why encrypt and then instantly decrypt?

Reply:
The encryptable is decrypted with the private_key of the receiver before this line. After receive, we want to decrypt the encryptable with the personal_team_password, same as all other encryptables. Because of that, we encrypt it first with the personal_team_password. But the #receive function is called from the #show function of the encryptablesController. The #show function decrypts normal encryptables, because of that, we need to return a decrypted encryptable.